// webpack-dts-bundle/dist/index.js
"use strict";
/*
 * Copyright (C) 2017 Shuma Yoshioka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
const dts = require("dts-bundle");
class DtsBundlePlugin {
    constructor(options) {
        this.options = options;
    }
    apply(compiler) {
        compiler.hooks.afterEmit.tap('webpack-dts-bundle', () => {
            dts.bundle({
                name: this.options.name,
                main: this.options.main,
                out: this.options.out,
                exclude: this.options.exclude,
                removeSource: this.options.removeSource,
                verbose: this.options.verbose,
            });
        });
    }
}
exports.default = DtsBundlePlugin;
